bot/src/utils/AirhornAudio.ts

```diff
@@ -1,157 +1,109 @@
-import {readFileSync} from "fs";
-import path from "path";
-import {Readable} from "stream";
-import {VoiceChannel, VoiceConnection} from "discord.js";
-import {config} from "./Configuration";
+import {
+  AudioPlayer,
+  AudioPlayerStatus,
+  createAudioPlayer,
+  createAudioResource,
+  entersState,
+  joinVoiceChannel,
+  StreamType,
+  VoiceConnection,
+  VoiceConnectionStatus,
+} from '@discordjs/voice';
+import { VoiceBasedChannel } from 'discord.js';
+import { log } from '../bot';
+import { addToCounterQueue, addToRepairGuildIdQueue, addToUserQueue, CounterQueueItem, UserQueueItem } from './QueueHandler';
 
-const audioBuffers: {
-  [key: string]: Buffer
-} = {};
+export const DEFAULT_AIRHORN_MAX_QUEUE_ITEMS = 3;
 
 type GuildQueueItem = {
-  voiceChannel: VoiceChannel,
-  soundFileName: string
+  soundFileReference: string;
+  counterQueueItem: CounterQueueItem;
+  userQueueItem: UserQueueItem;
 };
 
-const guildQueues = new Map<string, GuildQueueItem[]>();
+const guildQueues: Map<string, GuildQueueItem[]> = new Map();
 
-// Map of the sound name, with an inner map of the name of the variant and the sound file path
-const soundFiles = new Map<string, Map<string, string>>();
-// Map of the sound name, with the values being an array of the names of the variants for that sound
-export const soundVariants = new Map<string, string[]>();
-
-const soundsKeys = Object.keys(config.sounds);
-for (const soundKeysIndex in soundsKeys) {
-  const soundKey = soundsKeys[soundKeysIndex];
-  soundFiles.set(soundKey, new Map<string, string>());
-  soundVariants.set(soundKey, []);
-  const variants = config.sounds[soundKey].variants;
-  const variantsKeys = Object.keys(variants);
-  for (const variantKeysIndex in variantsKeys) {
-    const variantKey = variantsKeys[variantKeysIndex];
-    const variantFileName = variants[variantKey];
-    audioBuffers[soundKey + "/" + variantFileName] = readFileSync(path.join(__dirname, "..", "..", "sounds", soundKey, variantFileName));
-    soundFiles.get(soundKey)?.set(variantKey.toLowerCase(), soundKey + "/" + variantFileName);
-    soundVariants.get(soundKey)?.push(variantKey);
-  }
+export function getTotalItemsInGuildQueue(guildId: string): number {
+  return (guildQueues.get(guildId) || []).length;
 }
 
-function getGuildQueue(guildId: string): GuildQueueItem[] {
-  return guildQueues.get(guildId) || [];
+async function playSound(player: AudioPlayer, soundFileReference: string): Promise<unknown> {
+  const resource = createAudioResource(soundFileReference, {
+    inputType: StreamType.Arbitrary,
+  });
+  player.play(resource);
+  // Give it a maximum of 5 seconds to start playing, then another 5 seconds to finish
+  await entersState(player, AudioPlayerStatus.Playing, 5e3);
+  return entersState(player, AudioPlayerStatus.Idle, 5e3);
```

Comment:
Do we mean to be returning the promise here?  Or should this be an `await` and then a void return?

Reply:
Switched, the result isn't really needed, it just needs to wait until the sound has completed playing.